### DISCUSSION TITLE
从2.11升级到2.17 数据库起不来

### DISCUSSION THREAD

**Discussion by sicode** [halo-dev/halo#6520] [Q&A]
### 系统信息

Ubuntu 18.04 

### 使用的哪种方式运行？

Docker Compose

### 发生了什么？

太久没升级，遂按照文档中升级步骤，修改yaml中的镜像版本
重新启动就如日志

### 相关日志输出

```shell
HaloDB    | The files belonging to this database system will be owned by user "postgres".
HaloDB    | This user must also own the server process.
HaloDB    |
HaloDB    | The database cluster will be initialized with locale "en_US.utf8".
HaloDB    | The default database encoding has accordingly been set to "UTF8".
HaloDB    | The default text search configuration will be set to "english".
HaloDB    |
HaloDB    | Data page checksums are disabled.
HaloDB    |
HaloDB    | initdb: error: directory "/var/lib/postgresql/data" exists but is not empty
HaloDB    | initdb: hint: If you want to create a new database system, either remove or empty the directory "/var/lib/postgresql/data" or run initdb with an argument other than "/var/lib/postgresql/data".
```

**Comment by ruibaby** [comment]
建议提供完整的 docker compose 文件以及日志。

**Comment by sicode** [comment]
compose文件内容 

version: "3"

services:
  Halo:
    image: halohub/halo:2.17
    container_name: Halo
    restart: on-failure:3
    depends_on:
    -  HaloDB
    networks:
      halo_network:
    volumes:
      - /var/lib/docker/volumes/Halo:/root/.halo2
    ports:
      - "8090:8090"
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost:8090/actuator/health/readiness"]
      interval: 30s
      timeout: 5s
      retries: 5
    #  start_period: 30s
    command:
      - --spring.r2dbc.url=r2dbc:pool:postgresql://HaloDB/halo
      - --spring.r2dbc.username=halo
      # PostgreSQL 的密码，请保证与下方 POSTGRES_PASSWORD 的变量值一致。
      - --spring.r2dbc.password=********
      - --spring.sql.init.platform=postgresql
      # 外部访问地址，请根据实际需要修改
      - --halo.external-url=http://localhost:8090/
      # 初始化的超级管理员用户名
      #- --halo.security.initializer.superadminusername=admin
      # 初始化的超级管理员密码
      #- --halo.security.initializer.superadminpassword=********
  HaloDB:
    image: postgres:latest
    container_name: HaloDB
    restart: on-failure:3
    networks:
      halo_network:
    volumes:
      - /var/lib/docker/volumes/HaloDB:/var/lib/postgresql/data
    ports:
      - "5432:5432"
    healthcheck:
      test: [ "CMD", "pg_isready" ]
      interval: 10s
      timeout: 5s
      retries: 5
    environment:
      - POSTGRES_PASSWORD=********
      - POSTGRES_USER=halo
      - POSTGRES_DB=halo
      - PGUSER=halo

networks:
  halo_network:


至于日志就只有上面所发内容，数据库一直在重复相同的日志内容

**Comment by ruibaby** [comment]
```
image: postgres:latest
```

应该是因为 postgres 使用的 latest 标签，升级到最新版本之后无法使用旧数据。如果你之前是使用的文档提供的 Docker Compose 示例，可以将 postgres 镜像的标签固定为 `15.4` 试试，即 `postgres:15.4`

**Comment by sicode** [comment]
按指示改成15.4版本，依然是起不来，日志提示与之前一致

The files belonging to this database system will be owned by user "postgres".
This user must also own the server process.
The database cluster will be initialized with locale "en_US.utf8".
The default database encoding has accordingly been set to "UTF8".
The default text search configuration will be set to "english".
Data page checksums are disabled.
initdb: error: directory "/var/lib/postgresql/data" exists but is not empty
initdb: hint: If you want to create a new database system, either remove or empty the directory "/var/lib/postgresql/data" or run initdb with an argument other than "/var/lib/postgresql/data".
